### DISCUSSION TITLE
Hard disk icon in "Computer" -  what is it for?

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#1964] [Q&A] [answered]
In pcmanfm-qt in "Computer" there is always an  icon for the harddisk, but it's actually quite useless as you can do nothing with it, clicking you get "Can't Mount File", then "This path is not mounted"

![schermata-19-17-29](https://user-images.githubusercontent.com/10681413/112363636-033ab080-8cd6-11eb-8b95-4e28b714470a.png)

**Comment by tsujan** [comment, marked as answer]
It's what GVFS provides and pcmanfm-qt adds it as a bonus. We can do nothing more about it.

**Reply by stefonarch** [reply]
I never used it actually, there was just an user of Lubuntu who wanted to open his harddrive. Would be nice to show the partitions or something useful.

**Reply by tsujan** [reply]
I've made it hidden.

Tell that user that it's GVFS, not us ;)

**Reply by stefonarch** [reply]
Playing around I saw that everything in computer:/// was set to open with qdirstat and that you can set anything but it will always open in pcmanfm-qt, "open with" works as it should.

**Reply by tsujan** [reply]
BTW, I think we might be able to add mount to `computer:///`. I'll experiment with the idea but I don't think the result — if any — will be ready before 0.17.

**Reply by tsujan** [reply]
Strange that I see the opposite of what's described here.

I have 6 items inside `computer:///`. The first one, "File System", opens the root directory. The second one only shows my SSD, "SAMSUNG MZVLW128HEGR", which isn't something to mount; actually, my root directory is on it, which can be opened by "File System". The other items are my HDD partitions and all of them can be opened.

As for the USB stick, when I insert it, its icon appears in `computer:///` and when I click its icon, it gets mounted correctly.

So, I think there's nothing more to do: What can be mounted gets mounted and opened; what's already mounted can be opened.

The only item that can't be mounted in my case is "SAMSUNG MZVLW128HEGR". As https://wiki.gnome.org/Projects/gvfs/backends says, computer shows the list of devices connected to the PC. I can accept "SAMSUNG MZVLW128HEGR" as a piece of info. It's the same as "File System" when mounted.

@stefonarch, in your screenshot, there's a "JetFlash" item. If it's your USB stick, doesn't it get mounted when you click it?

**Reply by tsujan** [reply]
I tested Nautilus. It couldn't open any item under `computer:///`: except for "SAMSUNG MZVLW128HEGR", which created the same error message, nothing happened when I double clicked the other items that could be opened by pcmanfm-qt.

But Nautilus had an "Unmount" option in its right-click menu. We might be able to add it.

**Reply by stefonarch** [reply]
> @stefonarch, in your screenshot, there's a "JetFlash" item. If it's your USB stick, doesn't it get mounted when you click it?

yes, this works, disappears only after unplugging. But no partitions, there is one /boot and the rest is an encrypted LUKS volume with /var and /

**Reply by tsujan** [reply]
Consider them just as what GVFS puts there. Some can be mounted, some are just pieces of info.

I opened https://github.com/lxqt/libfm-qt/issues/658 and I think that's the only think we might be able to add.

**Reply by tsujan** [reply]
It was easier than I thought:

![unmount](https://user-images.githubusercontent.com/981076/112395729-95fe3e00-8d1c-11eb-8f89-8d1c885d5798.png)

**Reply by tsujan** [reply]
OK, having followed this idea to its logical conclusion, I succeeded in adding "Unmount" and "Eject" items to context menus when possible. For the first time, `computer:///` is usable in my older laptop because I can eject its empty DVD tray from there :)

Still not sure if it's a good idea to add this feature before 0.17 because it goes deep into libfm-qt's core.

**Reply by stefonarch** [reply]
> Still not sure if it's a good idea to add this feature before 0.17 because it goes deep into libfm-qt's core.

There is no hurry no?

**Reply by tsujan** [reply]
> There is no hurry no?

No. We need to test it without haste. I'll make the PR but won't merge it before 0.17.

Only if you opened this good page one month sooner... ;)

**Reply by tsujan** [reply]
BTW, do you know of any icon for the "mounted state" that exists in most icon sets? I found nothing good but I liked to experiment by adding an emblem when an item was mounted in `computer:///`.

EDIT: Since we recommend Qt-compatible icon sets like Breeze, I think "emblem-mounted" is the best option.

**Reply by stefonarch** [reply]
![schermata-13-38-47](https://user-images.githubusercontent.com/10681413/112474150-8cea8c80-8d6f-11eb-8521-aedf9050f977.png)
It does look weird for a disk, didn't find anything useful. Only option could be having this icon for unmounting at the right like in the left sidebar.

**Reply by tsujan** [reply]
I used "emblem-mounted" in https://github.com/lxqt/libfm-qt/pull/660 because it was in Breeze and Oxygen (and also in my icon set). I agree that they've used a weird shape in Breeze; it's a little better in Oxygen. It could be easily changed if a better icon is found.

I chose the top right corner because the other corners were already reserved by other kinds of elblems:

![emblem](https://user-images.githubusercontent.com/981076/112475606-a0f1b680-8d8e-11eb-8cd9-75707a060e7e.png)

**Reply by tsujan** [reply]
> Only option could be having this icon for unmounting at the right like in the left sidebar.

It's the "eject" icon, while this is about unmounting. But you're right: in Breeze it's better (in my set they're very similar — I'd thought of this situation a long time ago ;) )
